exec-wait.cpp

```diff
@@ -0,0 +1,109 @@
+#include <algorithm>
```

Comment:
a meta-question is whether the complexity of two languages here will be a problem. i don't think you need to rewrite your code (and eventually it would be nice to have everything in golang, for example) but i do worry a bit about it.

Reply:
Yeah, since the goal is to have a static executable to hook  into the image, I was weighing the complexity of adding C++ vs doing some sort of python compilation, and I thought C++ was simpler.  

We could potentially package this tool in a separate repo and pull it in and build it similar to the squashfs stuff.  It's a pretty isolated task, so i would not be concerned with replacing it with something else later if we want to do that.